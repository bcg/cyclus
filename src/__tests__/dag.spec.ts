import buildDAG from "../dag";

test("buildDAG should return correct built order", () => {
  expect(
    buildDAG(
      {
        a: "A",
        b: "B",
        c: "C",
        d: "D",
        e: "E",
        f: "F"
      },
      [["a", "d"], ["f", "b"], ["b", "d"], ["f", "a"], ["d", "c"]]
    )
  ).toMatchSnapshot();
});

test("build DAG should detect cycle", () => {
  expect(() =>
    buildDAG(
      {
        a: "A",
        b: "B",
        c: "C",
        d: "D",
        e: "E",
        f: "F"
      },
      [["a", "d"], ["f", "b"], ["b", "d"], ["f", "a"], ["d", "c"], ["c", "f"]]
    )
  ).toThrow("Cycle detected");
});
