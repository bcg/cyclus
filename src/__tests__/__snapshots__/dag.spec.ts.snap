// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildDAG should return correct built order 1`] = `
Array [
  "f",
  "a",
  "b",
  "d",
  "c",
  "e",
]
`;
